File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        Arrays.sort(citations);
        
        int n = citations.length;
        
        for (int i = 0; i < n; i++) {
            int h = n - i;   // 현재 위치 이상 논문 개수
            if (citations[i] >= h) {
                return h;
            }
        }
        
        return 0;
    }
}

File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    citations.sort()
    
    answer = 0
    for i in range(len(citations)):
        upTotal = len(citations[i:])   # 현재 값 이상인 논문 수
        
        if citations[i] >= upTotal:
            answer = upTotal
            break
                
    return answer